test/test_collection.py

```diff
@@ -32,6 +35,21 @@ def failed(self, event):
         pass
 
 class TestExplainableCollection(unittest.TestCase):
+    def _compare_command_dicts(self, ours, theirs):
```

Comment:
Does this method need to be recursive? What if we just compare the top-level command arguments exactly with self.assertEqual?

Reply:
Then you run into a bunch of issues with comparing SON objects and dictionaries. This makes it difficult because the document returned by the command listener has both SONs and dicts, and it's easiest to just compare the key value pairs recursively

Reply:
I expect that a SON would compare equal to a standard dictionary:
```
>>> SON() == {}
True
```

Reply:
It should, but I given that Python 3.5 has unordered dicts by default, I think it's better to be absolutely sure of the comparison.

Reply:
I'm don't fully understand why SON vs dict ordering matters here. Can you give an example test failure with my proposed approach?

Reply:
You're right. I will make this change.